packages/markup-frontstage/src/components/ui/MarkupFrontstageProvider.tsx

```diff
@@ -0,0 +1,349 @@
+    emphasizedElementsProps?: EmphasizeElementsProps
+  ) {
+    super();
+
+    this._isEditable = isEditable;
+    this._viewState = viewState;
+    this._iModelConnection = iModelConnection;
+
+    this._svg = svg;
+    this._viewElements = viewElements;
+    this._emphasizedElementsProps = emphasizedElementsProps;
+
+    this._markupLayout = new ContentLayoutDef({
+      descriptionKey: MarkupFrontstage.translate("contentLayoutDefDescription"),
+      priority: 50,
+      id: MarkupFrontstageConstants.VIEW_LAYOUT_ID,
+    });
+
+    this._contentGroup = new ContentGroup({
+      contents: [
+        {
+          id: MarkupFrontstageConstants.VIEW_LAYOUT_ID,
+          classId: contentControlClassId,
+          applicationData: {
+            viewState: this._viewState,
+            iModelConnection: this._iModelConnection,
+            rulesetId: this._rulesetId,
+          },
+        },
+      ],
+    });
+
+    FrontstageManager.onFrontstageReadyEvent.addListener(
+      this._onFrontstageReady
+    );
+    FrontstageManager.onFrontstageDeactivatedEvent.addListener(
+      this._onFrontstageDeactivated
+    );
+  }
+
+  /** Get Add Markup event. */
+  public readonly onAddMarkupEvent = new AddMarkupEvent();
+
+  /** Get Stop Markup event. */
+  public readonly onStopMarkupEvent = new StopMarkupEvent();
+
+  private _onCloseAsync = async () => {
+    this._onFrontstageChange();
+    this.onStopMarkupEvent.raiseEvent();
+    await FrontstageManager.closeNestedFrontstage();
+  };
+
+  private _onFrontstageDeactivated = async (
+    args: FrontstageDeactivatedEventArgs
+  ) => {
+    if (
+      args.deactivatedFrontstageDef.id ===
+      MarkupFrontstageConstants.FRONTSTAGE_ID
+    ) {
+      await this._stopMarkup();
+      FrontstageManager.onFrontstageDeactivatedEvent.removeListener(
+        this._onFrontstageDeactivated
+      );
+    }
+  };
+
+  private _onFrontstageReady = async (args: FrontstageReadyEventArgs) => {
+    if (args.frontstageDef.id === MarkupFrontstageConstants.FRONTSTAGE_ID) {
+      await this._startMarkup();
+    } else {
+      FrontstageManager.onFrontstageReadyEvent.removeListener(
+        this._onFrontstageReady
+      );
+    }
+  };
+
+  // call this whenever you need to persist the view from changing from one view to other.
+  private _onFrontstageChange() {
+    const newViewPort = IModelApp.viewManager.selectedView;
+    if (newViewPort) {
+      const oldViewState = newViewPort.view.clone();
+      const oldEmphasizedElementsProps = this._emphasizedElementsProps;
+      FrontstageManager.onFrontstageReadyEvent.addOnce(() => {
+        const selectedView = IModelApp.viewManager.selectedView;
+        if (selectedView && oldViewState) {
+          // change viewflags here if needed, should be before following.
+          selectedView.changeView(oldViewState);
+        }
+        // As per Diego, emphasized elements are not saved in view state, which we have to preserve manually as follow
```

Comment:
Should probably not reference anyone who is not a code owner in coments

Reply:
Sad to remove his, name wish I could refer him as IModelTeam or something that could go cool with :) well, now we know who to bug :)